ChronoVisor/src/KeeperRegistry.cpp

```diff
@@ -953,8 +961,36 @@ int KeeperRegistry::unregisterGrapherProcess(GrapherIdCard const& grapher_id_car
 
     RecordingGroup& recording_group = ((*group_iter).second);
 
-    std::stringstream id_string;
-    id_string << grapher_id_card;
+    // we are about to unregister the grapher so the group can't perform recording duties 
+    // if it were an active recordingGroup before remove it from rotation
+    if(recording_group.isActive())
+    {
+        auto active_group_iter = activeGroups.begin();
+        while (active_group_iter != activeGroups.end())
+        {
+            if((*active_group_iter) != &recording_group) 
+            { ++active_group_iter;}
+            else
+            { break; }
+        }
+
+        if(active_group_iter != activeGroups.end())
+        {
+            //INNA: what do we do with any active Stories that this group were recordng? force release them and notify clients?
```

Comment:
Since we do not have an API to initiate msgs to clients from Visor, we cannot notify clients for now. We may release the Stories that are being recording by this leaving Grapher and mark it in the ChronicleMetaDirectory. The following recording call with fail with a special error code.
At the same time, the Keepers in this being deactivated RecordingGroup should be able to start dumping all their data into local drives to avoid data loss while waiting for an active Grapher coming back in to re-activate themselves and this RecordingGroup.
I think it is rare to reach here. A Grapher sending unregister msg while all associated Keepers are still being actively recording events should not happen in normal situation. For system shutdown, Keepers should shutdown first. Then the RecordingGroup should become inactive when the Grapher unregisters itself. We will need to discuss something like a SOP for system shutdown and reboot as we did for the metadata operations.

Reply:
this is definitely not coming in this release.